src/torrent.cpp

```diff
@@ -8547,6 +8527,16 @@ namespace libtorrent
 		stop_announcing();
 	}
 
+	bool torrent::need_save_resume_data() const
+	{
+		// save resume data every 15 minutes regardless, just to
+		// keep stats up to date
+		return m_need_save_resume_data ||
+			// cast ratio<1l, 1000000000l> to seconds_t ratio
```

Comment:
I don't think the answer to that question belongs in the source code but rather as a reply to the question